### ISSUE TITLE
request:  suppress all output but errors

### ISSUE THREAD

**Issue by solshark** [invertase/melos#556]
### Is there an existing feature request for this?

- [X] I have searched the existing issues.

### Description

I'm running big project with 15+ packages (libs + apps) on CI server with gitlab-runner. Melos output is huge even for success path, so it would be awesome to silent all output except errors.

### Reasoning

It will be useful 'coz it will simplify CI logs analysis. Could you imagine CI pipeline running 'get' per package just pruning package name.
My output is XX thousands lines, 4-5 of them is simply notes like that:
```
[email_delivery_service]: + webview_flutter_platform_interface 2.4.0
[email_delivery_service]: + webview_flutter_wkwebview 3.7.3
[email_delivery_service]: + webview_windows 0.1.9 (0.3.0 available)
[email_delivery_service]: + win32 5.0.6
[email_delivery_service]: + win32_registry 1.1.1
[email_delivery_service]: + window_manager 0.3.5
```

Please consider flag to suppress all output.

**Comment by luanpotter** [comment]
I was about to make the very same suggestion!

## Rationale

This is part of Flame's output when running `melos analyze`:

<details>
<summary>Warning: large screenshot</summary>
<img src="https://github.com/user-attachments/assets/19ed3e04-0eca-40f5-8d2f-35ea81305c5a" />
</details>

I can't even screenshot the whole thing, due to how big it is. As @solshark mentioned, it can get a bit unhinged. Same for `melos format`, `melos run test`, etc.

[This is an example run of tests on CI](https://github.com/flame-engine/flame/actions/runs/9090500947/job/24983398949?pr=3113), it is quite hard to find the actual broken test.

## Proposal

My proposal would be to add a `--quiet` flag to commands like `analyze`, `format`, `run`, `exec`.

When this flag is present, "blocks" that had no errors or warnings whatsoever are completely omitted.

Luckily, melos already employs the concept of "blocks" in order to support concurrent execution.

In fact the code is already structured with three layers of abstraction:

![image](https://github.com/user-attachments/assets/a0562e81-b14b-49dd-9997-02badc902554)

* the command runner call log and group functions on a `MelosLogger` wrapper
* `MelosLogger` handles grouping and delegates printing to `Logger`
* `Logger` is the actual implementation that comes from `cli_utils`

In this case, we can easily hook up into the `MelosLogger` layer, as it already understands the necessary concepts of batching and grouping. We only would need to hold the concept of a "status" or "log level" for each group, and consider the `--quiet` flag in order to update the function `flushGroupBufferIfNeed`.

I believe the necessary changes would be quite simple all things considered (at least as a first step for `analyze` & `format` - `run`/tests could prove more difficult); and if this here Rationale & Proposal make sense, I would love to contribute with a PR to implement this. It would make parsing results both locally and on CI for our flame monorepo much easier!

**Comment by spydon** [comment]
@luanpotter sounds good, I'm thinking that it would be good if the quiet option could be global and not specific to commands in the end, and that it can also be specified directly in `melos.yaml`.

For the tests, that's why we used `spec` for that previously (which integrates with Melos), but we never figured out why that sometimes stopped working in the CI (most likely running out of memory).

**Comment by luanpotter** [comment]
I like that idea @spydon! I think it makes sense for it to be both an ad-hoc option and also allowed to be specified globally.

**Comment by ikaem-csq** [comment]
I wonder what is the situation with this idea?

**Comment by spydon** [comment]
> I wonder what is the situation with this idea?

Nobody has started working on it, if you're interested it is up for grabs. 

If you're in a *nix shell you should also be able to just direct stdout to `/dev/null`.

```sh
melos bootstrap > /dev/null
```

**Comment by spydon** [comment]
Can I assign you to this @luanpotter? :)

**Comment by luanpotter** [comment]
Yeah I can take a look when I have some time! I am really passionate about the feature just been busy 😭

**Comment by spydon** [comment]
This was finally fixed in v8.9.0!
